test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -197,6 +208,71 @@ def test_emacs(image):
     _run_cmd_on_container(container_name, ctx, "emacs -version")
 
 
+@pytest.mark.model("N/A")
+def test_cuda_paths(gpu):
+    """
+    Test to ensure directory structure for GPU files has cuda version in it
+
+    :param gpu: gpu image uris
+    """
+    image = gpu
+    if "example" in image:
+        pytest.skip("Skipping example containers which are not explicitly tied to a cuda version")
+
+    dlc_path = os.getcwd().split('/test/')[0]
+    job_type = "training" if "training" in image else "inference"
+
+    # Ensure that image has a supported framework
+    frameworks = ("tensorflow", "pytorch", "mxnet")
+    framework = ''
+    for fw in frameworks:
+        if fw in image:
+            framework = fw
+            break
+    assert framework, f"Cannot find any frameworks {frameworks} in image uri {image}"
+
+    # Get cuda and framework version through regex
+    cuda_version_match = re.search(r'-cu\d+-', image)
+    cuda_version = cuda_version_match.group().strip('-')
+
+    framework_version_match = re.search(r':\d+.\d+.\d+', image)
+    framework_version = framework_version_match.group().strip(':')
+
+    python_version_match = re.search(r'py\d+', image)
+    python_version = python_version_match.group()
+    framework_version_path = os.path.join(dlc_path, framework, job_type, 'docker', framework_version)
+    if not os.path.exists(os.path.join(framework_version_path, python_version)):
+        # Use the pyX version as opposed to the pyXY version if pyXY path does not exist
+        python_version = python_version[:3]
+
+    # Check buildspec for cuda version
+    buildspec = 'buildspec.yml'
+    if is_tf1(image):
+        buildspec = 'buildspec-tf1.yml'
+
+    cuda_in_buildspec = False
+    cuda_in_buildspec_ref = f"CUDA_VERSION {cuda_version}"
+    buildspec_path = os.path.join(dlc_path, framework, buildspec)
+    with open(buildspec_path, 'r') as bf:
+        for line in bf:
+            if cuda_in_buildspec_ref in line:
+                cuda_in_buildspec = True
+                break
+
+    try:
+        assert cuda_in_buildspec, f"Can't find {cuda_in_buildspec_ref} in {buildspec_path}"
+    except AssertionError as e:
+        if not is_dlc_cicd_context():
+            LOGGER.warn(f"{e} - not failing, as this is a(n) {os.getenv('BUILD_CONTEXT', 'empty')} build context.")
+        else:
+            raise
```

Comment:
Why would we not want this Error to be raised outside of cicd_context?

It seems like the sort of test that you would always want, and can easily skip if you're an "advanced" enough user that you're building your own images and running DLC tests.

Reply:
If you are testing older images locally, this will fail. Wanted to avoid causing unnecessary failures and warn instead

Reply:
Smart.